torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -959,18 +959,12 @@ def speculate_branch(branch):
             true_shared + unique_true + unique_false,
         )
 
-        flat_example_value = pytree.tree_map_only(
-            torch.fx.Proxy,
-            lambda a: a.node.meta["example_value"],
-            true_r.as_proxy(),
-        )
-
         return _call_function_and_unflatten_output(
             tx,
             torch.ops.higher_order.cond,
             p_args,
             {},
-            flat_example_value,
+            None,
```

Comment:
By not providing an example value, wrap_fx_proxy will call the hop with the fake inputs, which re-uses all the dispatch key implementations so that we don't need to have redundant code that merge the fake outputs in different places.

Reply:
nice